test/unit/SourceLocator/StringSourceLocatorTest.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace BetterReflectionTest\SourceLocator;
+
+use BetterReflection\Identifier\Identifier;
+use BetterReflection\Identifier\IdentifierType;
+use BetterReflection\SourceLocator\StringSourceLocator;
+use InvalidArgumentException;
+
+/**
+ * @covers \BetterReflection\SourceLocator\StringSourceLocator
+ */
+class StringSourceLocatorTest extends \PHPUnit_Framework_TestCase
+{
+    public function testInvokableLoadsSource()
+    {
+        $sourceCode = '<?php echo "Hello world!";';
+
+        $locator = new StringSourceLocator($sourceCode);
+
+        $locatedSource = $locator->__invoke(new Identifier(
+            'does not matter what the class name is',
+            new IdentifierType(IdentifierType::IDENTIFIER_CLASS)
+        ));
+
+        $this->assertSame($sourceCode, $locatedSource->getSource());
+        $this->assertNull($locatedSource->getFileName());
+    }
+
+    public function testConstructorThrowsExceptionIfEmptyStringGiven()
+    {
+        $this->setExpectedException(InvalidArgumentException::class);
+        new StringSourceLocator('');
```

Comment:
It would not make any sense to accept an empty string as source. Otherwise, nothing else will stop you doing it, and it just won't find the thing you're trying to reflect, and you'll be like "WHY DOES MY THING NOT FIND REFLECTION" and 3 hours later you find it's because you accidentally passed an empty string to the `StringSourceLocator` because of a misnamed variable or something, and then you'll be like

![1234931504682](https://cloud.githubusercontent.com/assets/496145/8542230/0937be5a-248c-11e5-8403-d09f7252f3e2.jpg)

However, I did forget to expect `EmptyPhpSourceCode` not `InvalidArgumentException` so I've pushed again.